skyportal/handlers/api/instrument.py

```diff
@@ -487,104 +496,137 @@ def put(self, instrument_id):
-                if not len(field_fov_attributes) == 2:
-                    return self.error(
-                        'If field_fov_type is rectangle, then should supply width and height for field_fov_attributes'
+                if not field_fov_type.lower() in ["circle", "rectangle"]:
+                    return self.error('field_fov_type must be circle or rectangle')
+                if isinstance(field_fov_attributes, list):
+                    field_fov_attributes = [float(x) for x in field_fov_attributes]
+                else:
+                    field_fov_attributes = [float(field_fov_attributes)]
+
+                center = SkyCoord(0.0, 0.0, unit='deg', frame='icrs')
+                if field_fov_type.lower() == "circle":
+                    if not len(field_fov_attributes) == 1:
+                        return self.error(
+                            'If field_fov_type is circle, then should supply only radius for field_fov_attributes'
+                        )
+                    radius = field_fov_attributes[0]
+                    regions = CircleSkyRegion(center=center, radius=radius * u.deg)
+                elif field_fov_type.lower() == "rectangle":
+                    if not len(field_fov_attributes) == 2:
+                        return self.error(
+                            'If field_fov_type is rectangle, then should supply width and height for field_fov_attributes'
+                        )
+                    width, height = field_fov_attributes
+                    regions = RectangleSkyRegion(
+                        center=center, width=width * u.deg, height=height * u.deg
                     )
-                width, height = field_fov_attributes
-                regions = RectangleSkyRegion(
-                    center=center, width=width * u.deg, height=height * u.deg
-                )
-            data['region'] = regions.serialize(format='ds9')
-
-        schema = Instrument.__schema__()
-        try:
-            schema.load(data, partial=True)
-        except ValidationError as exc:
-            return self.error(
-                'Invalid/missing parameters: ' f'{exc.normalized_messages()}'
-            )
-        self.verify_and_commit()
+                data['region'] = regions.serialize(format='ds9')
 
-        if field_data is not None:
-            if (field_region is None) and (field_fov_type is None):
+            schema = Instrument.__schema__()
+            try:
+                schema.load(data, partial=True)
+            except ValidationError as e:
                 return self.error(
-                    'field_region or field_fov_type is required with field_data'
+                    'Invalid/missing parameters: ' f'{e.normalized_messages()}'
                 )
 
-            if type(field_data) is str:
-                field_data = pd.read_table(StringIO(field_data), sep=",").to_dict(
-                    orient='list'
-                )
+            if 'name' in data:
```

Comment:
Maybe we do something similar to the spectrum handler here?

Reply:
https://github.com/skyportal/skyportal/blob/main/skyportal/handlers/api/spectrum.py#L811

Reply:
Thanks for the suggestion! I'll have a look at it tomorrow.